structure/hash.go

```diff
@@ -145,6 +156,11 @@ func (t *TxStructure) updateHash(key []byte, field []byte, fn func(oldValue []by
 
 // HLen gets the number of fields in a hash.
 func (t *TxStructure) HLen(key []byte) (int64, error) {
+	if t.ignoreHashMeta {
```

Comment:
If TiDB doesn't use `HLen`, how about remove all logic about it?

Reply:
I worry that someone could call `HLen` in future.

Reply:
`HLen` API is reasonable, you can find the structure package provides API similar to Redis, and Redis also has [HLen](https://www.tutorialspoint.com/redis/hashes_hlen.htm).
This package design is general, it can be used by outside projects.
I think it is better to provide an option instead of limit its ability to TiDB only.

Reply:
Without this PR, it means whether you use `HLen` or not, you pay for the cost.
With this PR, it means you can choose to pay for exactly what you want.
If you wipe out `HLen`, it means this functionality is missing, you don't have choice.

I prefer providing flexible, and let them do their choices.